### ISSUE TITLE
systemd-analyze security --offline=1 user@1000.service ==> Assertion 'u' failed at src/core/unit.c:155, function unit_has_name(). Aborting.

### ISSUE THREAD

**Issue by trentbuck** [systemd/systemd#29154]
### systemd version the issue has been seen with

254.1

### Used distribution

Debian 13 Trixie

### Linux kernel version used

6.1.0-11-amd64

### CPU architectures issue was seen on

x86_64

### Component

systemd-analyze

### Expected behaviour you didn't see

`systemd-analyze security --offline=1 x@.service` should behave similarly to
`systemd-analyze security --offline=1 x.service` and
`systemd-analyze security x@.service`.

### Unexpected behaviour you saw

```bash
$ systemd-analyze security --offline=1 user@1000.service
Assertion 'u' failed at src/core/unit.c:155, function unit_has_name(). Aborting.
Aborted (core dumped)
```

This seems to happen with any `x@.service`.
Giving a full path (`systemd-analyze security --offline=1 /lib/systemd/system/*.service`) is how I found it initially.
I tried giving an instance, and not giving a path (`user@.service`, `user@1000.service`) which did not help.
I can also reproduce this on Debian 12 also (`systemd 252 (252.12-1~deb12u1)`).  The only difference is the line number (155 vs 153).

### Steps to reproduce the problem

1. Run `systemd-analyze security --offline=1 user@1000.service`.
2. Did it segfault?

**Comment by yuwata** [comment]
I cannot reproduce the issue with the current git HEAD.
It seems already fixed.

**Comment by yuwata** [comment]
Already fixed by f660c7fa56b247c278fdb2ebcfea37912f249524.
